chainer/functions/normalization/local_response_normalization.py

```diff
@@ -54,86 +58,129 @@ def check_type_forward(self, in_types):
+                 scale=None, indexes=None, unit_scale=None):
+        self.n = n
+        self.k = k
+        self.alpha = alpha
+        self.beta = beta
+        self._use_ideep = use_ideep
+
+        self.scale = scale
+        self.indexes = indexes
+        self.unit_scale = unit_scale
+
+    def forward_cpu(self, inputs):
         if self._use_ideep:
-            return self._backward_ideep(x, gy)
+            return self._backward_ideep(inputs)
 
+        x, y, gy = inputs
         half_n = self.n // 2
-        summand = self.y * gy[0] / self.unit_scale
+        summand = y * gy / self.unit_scale
         sum_part = summand.copy()
         for i in six.moves.range(1, half_n + 1):
             sum_part[:, i:] += summand[:, :-i]
             sum_part[:, :-i] += summand[:, i:]
 
-        gx = gy[0] * self.scale - 2 * self.alpha * self.beta * x[0] * sum_part
+        gx = gy * self.scale - 2 * self.alpha * self.beta * x * sum_part
         return gx,
 
-    def _backward_ideep(self, x, gy):
+    def _backward_ideep(self, inputs):
+        x, y, gy = inputs
+
         param = intel64.ideep.localResponseNormalizationParam(
             self.n, self.k, self.n * self.alpha, self.beta,
             intel64.ideep.localResponseNormalizationParam.lrn_across_channels
         )
         gx = intel64.ideep.localResponseNormalization.Backward(
-            intel64.ideep.array(x[0]),
-            intel64.ideep.array(gy[0]),
+            intel64.ideep.array(x),
+            intel64.ideep.array(gy),
             self.indexes,
             param)
         return gx,
 
-    def forward_gpu(self, x):
-        self.y = cuda.cupy.square(x[0])  # temporary
-        self.scale = cuda.cupy.empty_like(self.y)
-        _cu_conv_sum(self.scale, self.y, self.n)
-        cuda.elementwise(
-            'T x, T k, T alpha, T beta',
-            'T y, T scale',
-            '''scale = k + alpha * scale;
-               y = x * pow(scale, -beta);''',
-            'lrn_fwd')(x[0], self.k, self.alpha, self.beta,
-                       self.y, self.scale)
-        return self.y,
-
-    def backward_gpu(self, x, gy):
+    def forward_gpu(self, inputs):
+        x, y, gy = inputs
         summand = cuda.elementwise(
             'T scale, T y, T gy', 'T summand',
             'summand = y * gy / scale',
-            'lrn_bwd_summand')(self.scale, self.y, gy[0])
-        gx = cuda.cupy.empty_like(x[0])
+            'lrn_bwd_summand')(self.scale, y, gy)
+        gx = cuda.cupy.empty_like(x)
         _cu_conv_sum(gx, summand, self.n)
         cuda.elementwise(
             ' T x, T gy, T scale, T beta, T coeff', 'T gx',
             'gx = pow(scale, -beta) * gy - coeff * x * gx',
-            'lrn_bwd')(x[0], gy[0], self.scale,
+            'lrn_bwd')(x, gy, self.scale,
                        self.beta, 2 * self.alpha * self.beta, gx)
         return gx,
 
+    def backward(self, indexes, grad_outputs):
+        raise NotImplementedError
```

Comment:
Let me confirm that there is no essential reason to leave its double backward not implemented?

Reply:
Thank you for your review, again. Well, I put `NotImplementedError` here just because there was no trivial way to implement double-backward for this function.
If there is more appropriate way, I would like to follow such method.

Reply:
Thanks, then would you leave a comment as such because, I think, it is even valuable information?
``` Python
# No trivial way to implement double-backward for this function.
raise NotImplementedError
```

Reply:
Thank you for advice. I added the comment.